app/controllers/api_controller/providers.rb

```diff
@@ -5,7 +5,7 @@ module Providers
     CREDENTIALS_ATTR  = "credentials"
     AUTH_TYPE_ATTR    = "auth_type"
     DEFAULT_AUTH_TYPE = "default"
-    ENDPOINT_ATTRS    = %w(hostname ipaddress port)
+    ENDPOINT_ATTRS    = %w(hostname ipaddress port security_protocol).freeze
```

Comment:
I think the `app/models/provider.rb` is not relevant here.

The problem is that `Provider` and `ExtManagementSystem` class each delegate different set of attributes to a default endpoint. Further, default endpoint of a Provider is just first default endpoint of any ExtManagementSystems assigned to the `Provider`.

More importantly, the '/api/providers' manages only `ExtManagementSystem` records. We may want to add support for `Provider` records. Do we have a ticket for this in backlog somewhere?

Currently, there are only three sub-classes of the `Provider` class.
```
$ git grep '<\s*[:]*Provider'
app/models/manageiq/providers/ansible_tower/provider.rb:class ManageIQ::Providers::AnsibleTower::Provider < ::Provider
app/models/manageiq/providers/foreman/provider.rb:class ManageIQ::Providers::Foreman::Provider < ::Provider
app/models/manageiq/providers/openstack/provider.rb:class ManageIQ::Providers::Openstack::Provider < ::Provider
```